src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -133,6 +135,11 @@ private static Solution GetSolutionWithReplacedDocuments(Solution solution, IDoc
         /// </summary>
         public readonly Solution Solution;
 
+        /// <summary>
+        /// Tracing object that can be used to log information about the status of requests.
+        /// </summary>
+        public readonly TraceSource? TraceSource;
```

Comment:
I wonder if instead of directly exposing the trace source, we should just have an [ILogger](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/ILogger.cs,12) implementation (or similar, if the API for ILogger isn't quite right) that is created from a trace source.  Then each server creates it's logger on startup and passes it in to each request